### PR TITLE
Fast/compost

### PR DESCRIPTION
**Pull request by Mekle001**
# Changes proposed in this pull request:
- Accelerate barrels based on Primary Skill level, similar to the way we accelerate furnaces.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIUsesFurnace.java [hunk 1]
```diff
@@ -231,7 +231,7 @@ private IAIState accelerateFurnaces()
                 }
             }
         }
-        return getState();
+        return null;
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIUsesFurnace.java:234]: Is this even valid?
This should probably return some same default state.

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/composter/EntityAIWorkComposter.java [hunk 2]
```diff
@@ -92,6 +97,35 @@ public EntityAIWorkComposter(@NotNull final JobComposter job)
...
+                    {
+                        if (barrel.checkIfWorking() && !barrel.isDone())
+                        {
+                            barrel.tick();
+                        }
+                    }
+                }
+            }
+        }
+        return null;
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/composter/EntityAIWorkComposter.java:125]: Again return some same default state here.

**Review by marchermans — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIUsesFurnace.java (hunk 1 above)
Comment (Mekle001) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIUsesFurnace.java:234]: Actually, there is a significant behavior difference between null and getState. Returning a valid state here stops further execution on that tick, where returning null allows further execution. I can add comments here, though I found that documentation in AbstractEntityAIBasic

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/composter/EntityAIWorkComposter.java (hunk 2 above)
Comment (Mekle001) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/composter/EntityAIWorkComposter.java:125]: See previous comments

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIUsesFurnace.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIUsesFurnace.java:234]: @someaddons can probably confirm

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIUsesFurnace.java:234]: ye null state return means no action, not null means we traverse into that state and do not check further transitions of this state in the current tick. I thought about changing it, but back then this was needed for backwards compat due to the way our AI is structured/used.

**Review by marchermans — APPROVED** [review]
Oke for now this is fine, we keep track of the `null` option in #6028, cause that is a major codesmell and should be avoided at all costs.

**Pull request merged**